### PR TITLE
File generator

### PR DESCRIPTION
**Pull request by slavapa**
Please review my initial commits for file generation utility

### CODE REVIEW

File: file_utils.gemspec [hunk 1]
```diff
@@ -9,7 +9,10 @@ Gem::Specification.new do |s|
   s.email       = 'kolmanv@gmail.com'
   s.homepage    = 'http://github.com/kolmanv/bbfs'
   s.files       = ['lib/file_utils.rb',
-                   'lib/file_utils/file_utils.rb']
+                   'lib/file_utils/file_utils.rb',
+                   'file_utils/file_generator/file_generator.rb',
```
Comment (kolmanv) [inline file_utils.gemspec:13]: add 'lib' to all paths.

Did you try building the gem? Installing? Using?
Using gem means the gem itself and not the code directly.

File: lib/file_utils/file_generator/file_generator.rb [hunk 2]
```diff
@@ -0,0 +1,140 @@
+require 'fileutils'
+require_relative 'file_generator_parameters'
+require 'digest'
+
+
+#module BBFS
```
Comment (kolmanv) [inline lib/file_utils/file_generator/file_generator.rb:6]: Remove comment. In both lines.

File: lib/file_utils/file_generator/file_generator.rb [hunk 3]
```diff
@@ -0,0 +1,140 @@
+require 'fileutils'
+require_relative 'file_generator_parameters'
```
Comment (kolmanv) [inline lib/file_utils/file_generator/file_generator.rb:2]: We don't use require relative.
add:
require 'file_utils/file_generator/file_generator_parameters'

Order requires in alphabetical order.

File: lib/file_utils/file_generator/file_generator.rb [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        end
+      end
+
+    end
+
+#  end
+#end
+
+
+#puts Time.now
```
Comment (kolmanv) [inline lib/file_utils/file_generator/file_generator.rb:131]: What are those comments, please remove.

File: lib/file_utils/file_generator/file_generator_parameters.rb [hunk 5]
```diff
@@ -0,0 +1,188 @@
+require 'yaml'
+
+#module BBFS
```
Comment (kolmanv) [inline lib/file_utils/file_generator/file_generator_parameters.rb:3]: remove comments.

File: test/file_utils/file_generator.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
```
Comment (kolmanv) [inline test/file_utils/file_generator.rb:0]: What is this file for?

**Comment by kolmanv** [comment]
Why don't you use the existing Params gem?
Please do use it.
People are working for the Params to be loaded from yml file.
Why did not you talk to me about design? We could save much of your work.

File: file_utils.gemspec (hunk 1 above)
Comment (slavapa) [inline file_utils.gemspec:13]: Kolman,

I need your help to resolve this.

**Comment by slavapa** [comment]
Koman,
I just did the initial release.
Now we can change to what we want.
We have to talk about this.

Regards,
Slava

-----Original Message-----

File: file_utils.gemspec (hunk 1 above)
Comment (slavapa) [inline file_utils.gemspec:13]: Kolman,

I need you help to resolve this issue.

Thanks,
Slava.

Git Hub Reference:
add 'lib' to all paths.

Did you try building the gem? Installing? Using?
Using gem means the gem itself and not the code directly.

-----Original Message-----

Comment (slavapa) [inline file_utils.gemspec:13]: Added  'lib' to all paths.

File: lib/file_utils/file_generator/file_generator.rb (hunk 3 above)
Comment (slavapa) [inline lib/file_utils/file_generator/file_generator.rb:2]: Fixed

File: lib/file_utils/file_generator/file_generator.rb (hunk 2 above)
Comment (slavapa) [inline lib/file_utils/file_generator/file_generator.rb:6]: Fixed. Removed comments

File: lib/file_utils/file_generator/file_generator.rb (hunk 4 above)
Comment (slavapa) [inline lib/file_utils/file_generator/file_generator.rb:131]: Fixed. Removed comments

File: test/file_utils/file_generator.rb (hunk 6 above)
Comment (slavapa) [inline test/file_utils/file_generator.rb:0]: This the the sitra arha or Klipot.
I deleted this file.

File: lib/file_utils/file_generator/file_generator.rb [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+      def get_sleep_time_in_seconds
+        @file_gen_params.sleep_time_in_seconds
+      end
+
+      def is_gen_file(file_counter)
+        if file_counter == 0 && @file_gen_params.is_tot_files_in_dir_random  then
+          @file_gen_params.total_files_in_dir =
+              rand(@file_gen_params.upper_limit_4_files_in_dir -
+                       @file_gen_params.lower_limit_4_files_in_dir) +
```
Comment (kolmanv) [inline lib/file_utils/file_generator/file_generator.rb:90]: Fix indentation.

**Pull request merged**